catroid/src/main/java/org/catrobat/catroid/io/StorageHandler.java

```diff
@@ -1370,4 +1372,113 @@ public void updateCodefileOnDownload(String projectName) {
+	 *
+	 * @param srcPath Path to either file or directory
+	 */
+	public void deleteAllFiles(String srcPath) {
+		File toDelete = new File(srcPath);
+
+		if (toDelete.isDirectory()) {
+			for (String file : toDelete.list()) {
+				deleteAllFiles(file);
+			}
+		}
+
+		toDelete.delete();
+	}
+
+	/**
+	 * Copies all files within the given source path into the given destination path.
+	 *
+	 * @param srcDirectoryPath Path to source directory.
+	 * @param dstDirectoryPath Path to destination directory. This gets created if it does not exist yet.
+	 * @return true if all fies were copied successfully, false otherwise.
+	 */
+	public static void copyRecursive (String srcDirectoryPath, String dstDirectoryPath) throws IOException {
+		File srcDirectory = new File(srcDirectoryPath);
+		if(!srcDirectory.exists()) {
+			throw new IOException("Cannot create copy of non existent file.");
+		}
+
+		if (srcDirectory.isDirectory()) {
+			for (String subDirectory : srcDirectory.list()) {
+				copyRecursive(srcDirectoryPath.concat(subDirectory), dstDirectoryPath.concat(subDirectory));
+			}
+		}
+
+		copyFile(srcDirectoryPath, dstDirectoryPath);
+	}
+
+	/**
+	 * Creates a copy of a file in the same directory.
+	 * @param srcFilePath Path to the original file.
+	 * @return Copied File or referenced original File depending on usage count handling.
+	 */
+	public static File copyFile(String srcFilePath) throws IOException {
+		return copyFile(srcFilePath, new File(srcFilePath).getParent());
+	}
+
+	/**
+	 * Creates a copy of a file in the specified destination directory (will be created if non existent).
+	 *
+	 * @param srcFilePath Path to the original file.
+	 * @param dstFileDirectory Destination directory for the copied file.
+	 * @return Copied File or referenced original File depending on usage count handling.
+	 */
+	public static File copyFile(String srcFilePath, String dstFileDirectory)
+			throws IOException {
+		File original = new File(srcFilePath);
+
+		if (!original.exists()) {
+			throw new IOException("Cannot copy file: SourceFile does not exist");
+		}
+
+		File copy = null;
+		try {
+			File destinationDir = new File(dstFileDirectory);
+			destinationDir.mkdirs();
+
+			copy = getFileCopyUniqueName(original.getName(), dstFileDirectory);
+			Files.copy(original, copy);
+		} catch (Exception e) {
+			if (copy != null && copy.exists()) {
+				copy.delete();
+			}
+			throw e;
+		}
+
+		return copy;
+	}
+
+	private static synchronized File getFileCopyUniqueName(String originalName, String dstDirectory){
+		int extensionStartIndex = originalName.lastIndexOf(".");
+		String extension = originalName.substring(extensionStartIndex);
+		String fileName = originalName.substring(0, extensionStartIndex);
+
+		int appendix = 0;
+
+		while(true) {
+			String copyFileName = fileName + COPY_FILENAME_POSTFIX + String.valueOf(appendix) + extension;
+			File copyFile = new File(dstDirectory, copyFileName);
+			if(!copyFile.exists()) {
+				return copyFile;
```

Comment:
I am looking for a unique file with appendix "_copy###" where ### represetnts ascending numbers.. so upon iterating from 0 upwards the first free file is returned, i.e. this is a unique file name within that folder